File: Dsa/weekly-practice/practiceday3.py
def Palindrome(arr):
    left=0
    right=len(arr)-1
    while left<right:
        if arr[left] != arr[right]:
            return False
        left+=1
        right-=1
    return True


def PalindromeForLoop(arr):
    for i in range(len(arr)//2):
        if arr[i] != arr[len(arr)-1-i]:
            return False

    return True
print(Palindrome("madam"))
print(Palindrome("madams"))
print(PalindromeForLoop("madam"))
print(PalindromeForLoop("madams"))

File: Dsa/weekly-practice/practiceday3.js
function Palindrome(arr){
    let left=0
    let right=arr.length-1
    while(left<right){
        if(arr[left]!=arr[right]){
            return false
        }
        left++
        right--
    }
    return true
}

function PlaindromeForLoop(arr){
    for(let i=0; i<arr.length/2;i++){
        if(arr[i] !=arr[arr.length-1-i]){
            return false
        }
    }
    return true
}
console.log(Palindrome("madam"))
console.log(PlaindromeForLoop("madams"))